scripts/zf_make_official_srpm

```diff
@@ -9,89 +9,88 @@ log()  { err "$me: $*"; }
 fail() { log "$*"; exit 1; }
 try()  { "$@" || fail "FAILED: $*"; }
 
-
-usage_msg() {
-  err
-  err "usage:"
-  err "  $me <tcpdirect-tarball>"
-  err
-  err "options:"
-  err
-}
-
+my_dir=$(cd "$(dirname "$0")" && /bin/pwd)
+top_dir=$(dirname "${my_dir}")
+
+tarball_files_from="
+versions.yaml
+Makefile
+Makefile.onload
+Makefile-top.inc
+doc
+src
+scripts/debian-templ
+scripts/tcpdirect_misc/tcpdirect-extract-notes
+scripts/zf_debug
+scripts/zf_install
+scripts/zf_make_official_deb
+scripts/zf_make_official_srpm
+scripts/zf_make_tarball
+scripts/zf_mkdist
+scripts/zf_uninstall
+"
 
 usage() {
-  usage_msg
+  echo
+  echo "usage:"
+  echo "  $(basename "$0") [options]"
+  echo
+  echo "options:"
+  echo "  --version <version>"
+  echo
   exit 1
 }
 
-
-logtry() {
-  log "$*"
-  try "$@"
-}
-
-
 cleanup() {
-  [ -d "$tmpd" ] && rm -rf "$tmpd"
+  echo "Deleting ${tmpdir}"
+  rm -rf "${tmpdir}"
 }
 
-
 ######################################################################
 # main()
+
 version=
-tarball=
 
 while [ $# -gt 0 ]; do
-    case "$1" in
-        --version)  shift; version="$1";;
-        -*)  usage;;
-        *)  tarball="$1";;
-    esac
-    shift
+  case "$1" in
+    --version)  shift; version="$1";;
+    *)  usage;;
+  esac
+  shift
 done
 
-if [ -z "$version" ]; then
+if [ -z "${version}" ]; then
     version=$(git rev-parse HEAD)
 fi
 
-# [ $# = 1 ] || usage
-# tarball="$1"
+tmpdir=$(mktemp -d "/tmp/${me}.XXXXXXXX")
+zf_prefix="tcpdirect-${version}"
+zf_spec="scripts/tcpdirect_misc/tcpdirect.spec"
+zf_tarball="${tmpdir}/${zf_prefix}.tar"
+zf_tgz="${tmpdir}/${zf_prefix}.tgz"
 
-tmpd=
 trap cleanup EXIT
-tmpd=$(mktemp -d /tmp/$me.XXXXXXXX)
-
-try [ -f "$tarball" ]
-tarballdir=$(cd "$(dirname "$tarball")" && /bin/pwd)
-tarballbase=$(basename "$tarball")
-tarball="$tarballdir/$tarballbase"
-try [ -f "$tarball" ]
-
-original_wd=$(/bin/pwd)
-try cd "$tmpd"
-
-# Edit the "Packager" line in the rpm spec file and rebuild tarball.
-try tar xf "$tarball"
-distd="$(basename "${tarball%.tgz}")"
-try [ -d "$distd" ]
-tab=$(echo -n -e "\t")
-try rm "$distd/scripts/zf_install" "$distd/scripts/zf_uninstall"
-
-try sed -i \
- -e "s/pkgversion-DEFINITION/%define pkgversion $version/g" \
- "$distd/scripts/tcpdirect.spec"
-
-tarball=$(basename "$tarball")
-try tar czf "$tarball" --owner=root --group=root "$distd"
-
-# Build source rpm.
-try mkdir -p rpmbuild/{BUILD,BUILDROOT,RPMS,SOURCES,SPECS,SRPMS}
-try rpmbuild --nodeps --define "_topdir $PWD/rpmbuild" -ts "$tarball"
-srpm=$(/bin/ls rpmbuild/SRPMS/*.rpm)
-srpmbase=$(basename "$srpm")
-destfile="$original_wd/$(echo "$srpmbase" | sed -e 's/\.el6//')"
-[ -f "$destfile" ] &&
-  fail "ERROR: '$destfile' already exists.  I will not overwrite."
-try cp "$srpm" "$destfile"
-echo "$destfile"
+
+# Create a tarball in tmpdir.
+# shellcheck disable=SC2086
+try tar -c -C "${top_dir}" --transform "s,^,${zf_prefix}/," -f "${zf_tarball}" -- ${tarball_files_from}
```

Comment:
Do these need to have shellcheck disabled?

Reply:
`shellcheck` suggests double quotes around `${tarball_files_from}` we don't want them:

```console
$ foo="
> Makefile
> Makefile.onload
> "

$ ls "${foo}"
ls: cannot access ''$'\n''Makefile'$'\n''Makefile.onload'$'\n': No such file or directory

$ ls ${foo}
Makefile  Makefile.onload
```

I believe that with double quotes, the variable is interpreted as one parameter/file.

One alternative would be using `-T`/`--files-from` with the list in a file, or, even better, `git-archive`, but it seems to require patching the CI pipelines. I've put a comment with a suggestion about `git-archive` and test.groovy in ON-15946.